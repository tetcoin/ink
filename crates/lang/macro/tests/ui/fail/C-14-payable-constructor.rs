use pro_lang as pro;

#[pro::contract]
mod noop {
    #[pro(storage)]
    pub struct Noop {}

    impl Noop {
        #[pro(constructor, payable)]
        pub fn abi_constructor() -> Self {
            Self {}
        }

        #[pro(message)]
        pub fn noop(&self) {}
    }
}

fn main() {}
