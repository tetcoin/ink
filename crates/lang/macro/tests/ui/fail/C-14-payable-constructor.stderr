error: encountered conflicting pro! attribute argument
 --> $DIR/C-14-payable-constructor.rs:9:28
  |
9 |         #[pro(constructor, payable)]
  |                            ^^^^^^^

error: constructors are implicitly payable
 --> $DIR/C-14-payable-constructor.rs:9:28
  |
9 |         #[pro(constructor, payable)]
  |                            ^^^^^^^
